packages/tooling/fast-tooling/src/web-components/html-render/html-render.ts

```diff
@@ -0,0 +1,275 @@
+                e.data.activeNavigationConfigId !== this.navigationConfigId
+            ) {
+                if (e.data.action === MessageSystemNavigationTypeAction.update) {
+                    const dataId: string = e.data.activeDictionaryId;
+                    const el: HTMLElement = this.shadowRoot.querySelector(
+                        "[" + this.dataDictionaryAttr + "=" + dataId + "]"
+                    );
+                    if (el) {
+                        this.currentElement = el;
+                        this.updateLayers(ActivityType.click, dataId, el);
+                    }
+                }
+            }
+        }
+    };
+
+    private updateLayers(
+        activityType: ActivityType,
+        dictionaryId: string,
+        elementRef: HTMLElement
+    ) {
+        if (this.renderLayers) {
+            this.renderLayers.forEach(value => {
+                value.elementActivity(activityType, dictionaryId, elementRef);
+            });
+        }
+    }
+
+    /// Mouse Handlers
+
+    public hoverHandler(e: MouseEvent): boolean {
+        const el: HTMLElement = e.composedPath()[0] as HTMLElement;
+        const dataId = el.getAttribute(this.dataDictionaryAttr);
+        if (
+            dataId !== null &&
+            !(
+                this.currentElement &&
+                dataId === this.currentElement.getAttribute(this.dataDictionaryAttr)
+            )
+        ) {
+            this.updateLayers(ActivityType.hover, dataId, el);
+        }
+        return false;
+    }
+
+    public blurHandler(e: MouseEvent): boolean {
+        this.updateLayers(ActivityType.blur, "", null);
+        return false;
+    }
+
+    private selectElement(el: HTMLElement, dataId: string) {
+        this.messageSystem.postMessage({
+            type: MessageSystemType.navigation,
+            action: MessageSystemNavigationTypeAction.update,
+            activeDictionaryId: dataId,
+            activeNavigationConfigId: this.navigationConfigId,
+        });
+        this.currentElement = el;
+        this.updateLayers(ActivityType.click, dataId, el);
+    }
+
+    private clearElement() {
+        this.messageSystem.postMessage({
+            type: MessageSystemType.navigation,
+            action: MessageSystemNavigationTypeAction.update,
+            activeDictionaryId: "",
+            activeNavigationConfigId: this.navigationConfigId,
+        });
+        this.currentElement = null;
+        this.updateLayers(ActivityType.clear, "", null);
+    }
+
+    public clickHandler(e: MouseEvent): boolean {
+        const el: HTMLElement = e.composedPath()[0] as HTMLElement;
+        const dataId = el.getAttribute(this.dataDictionaryAttr);
+        if (dataId !== null) {
+            this.selectElement(el, dataId);
+            e.stopPropagation();
+            return false;
+        }
+    }
+
+    public keyUpHandler(e: KeyboardEvent): boolean {
+        if (e.key === "Tab") {
```

Comment:
You should be able to import keycodes, such as keyCodeTab from `@microsoft/fast-web-utilities`, these correspond to `keyCode` so this can be `e.keyCode === keyCodeTab`.

Reply:
It seems that keyCode has been deprecated in favor of key.  https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/KeyboardEvent
I tried to switch it to keyCode but it is no longer possible to set this property using the KeyboardEventInit structure when creating new events using KeyboardEvent().

Reply:
Ah, I forgot that was an issue. Our React package masks it because it just includes the deprecated keyCode.